### PR TITLE
sc-20969- retry oidc auth on errors

### PR DESCRIPTION
**Pull request by avichayJit**
SC-20969 - retrying oidc auth on some types of errors

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -25,23 +25,43 @@ runs:
...
+            echo "has_tokens=1" >> $GITHUB_OUTPUT
+          elif [ "$response_code" -eq 503 ] || [ "$response_code" -eq 504 ]; then
+            # service unavailable, gateway timeout
+            echo "Response code is retryable - retrying"
+            exit 1
+          fi
+
+    - name: Verify we have tokens
       shell: bash
+      if: steps.oidc-auth.has_tokens == 0
```
Comment (yardensade) [inline action.yml]: You miss the "outputs" key like so:
`if: steps.set-cache-file.outputs.cache_file != ''`

File: action.yml [hunk 2]
```diff
@@ -25,23 +25,43 @@ runs:
...
-                "target_repo": "${{ fromJSON(github.event.inputs.client_payload).payload.full_repo_path }}",
-                "jit_event_id": "${{ fromJSON(github.event.inputs.client_payload).payload.jit_event_id }}",
-                "execution_id": "${{ fromJSON(github.event.inputs.client_payload).payload.execution_id }}"
-          }' \
-          ${{ fromJSON(github.event.inputs.client_payload).payload.jit_base_api }}/github/auth
-        )
-        
-        echo "github_token=$(echo $AUTH_RESPONSE | jq -r '.github_token')" >> $GITHUB_OUTPUT
-        echo "jit_token=$(echo $AUTH_RESPONSE | jq -r '.jit_token')" >> $GITHUB_OUTPUT
+        exit 99
```
Comment (yardensade) [inline action.yml:64]: why 99 and not 1?

File: action.yml [hunk 3]
```diff
@@ -146,7 +166,7 @@ runs:
       run: docker pull ${{ inputs.security_control }}
 
     - name: Run The Action
-      if: always()
+      if: always() && steps.oidc-auth.has_tokens == 1
```
Comment (yardensade) [inline action.yml]: missing outputs key

File: action.yml [hunk 4]
```diff
@@ -173,7 +193,7 @@ runs:
         mount_original_repo_command: ${{ fromJSON('["","-v $(pwd)/code:/code"]')[((!(fromJSON(github.event.inputs.client_payload).payload.runner_setup) || (fromJSON(github.event.inputs.client_payload).payload.runner_setup.checkout == true)))] }}
 
     - name: Set image id
-      if: always()
+      if: always() && steps.oidc-auth.has_tokens == 1
```
Comment (yardensade) [inline action.yml]: missing outputs key

File: action.yml (hunk 2 above)
Comment (adiattlan) [reply action.yml:64]: wanted an error code that we can identify when we see message on the resource management channel

Comment (avichayJit) [reply action.yml:64]: wanted an error code that we can identify when we see message on the resource management channel

**Review by yardensade — APPROVED** [review]

File: action.yml (hunk 1 above)
Comment (avichayJit) [reply action.yml]: yap -fixed

**Review by yardenQA — APPROVED** [review]

**Review by yardensade — APPROVED** [review]

**Pull request merged**